GAE/src/org/akvo/flow/rest/ReportRestService.java

```diff
@@ -0,0 +1,212 @@
+/*
+ *  Copyright (C) 2018 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.rest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import org.akvo.flow.dao.ReportDao;
+import org.akvo.flow.domain.persistent.Report;
+import org.akvo.flow.rest.dto.ReportDto;
+import org.akvo.flow.rest.dto.ReportPayload;
+import org.akvo.flow.servlet.ReportServlet;
+import org.springframework.beans.BeanUtils;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.waterforpeople.mapping.app.web.rest.dto.RestStatusDto;
+
+import com.gallatinsystems.user.dao.UserDao;
+
+
+/*
+ * @startuml
+ * UI -> ReportRestService: POST
+ * ReportRestService --> UI: Report with id (JSON object in QUEUED state)
+ * ReportRestService -> ReportServlet: START
+ * ReportServlet -> FlowService: POST with Report id
+ * FlowService -> ReportRestService: PUT Report in IN_PROGRESS state
+ * UI -> ReportRestService: GET
+ * ReportRestService --> UI: List of Reports by current user (JSON list)
+ * FlowService -> ReportRestService: PUT Report (FINISHED_SUCCESS state and filename)
+ * UI -> ReportRestService: GET
+ * ReportRestService --> UI: List of Reports by current user (JSON list)
+ * UI -> FlowService: GET filename
+ * FlowService --> UI: Report file
+ * @enduml
+ */
+
+@Controller
+@RequestMapping("/reports")
+public class ReportRestService {
+    private static final Logger log = Logger.getLogger(ReportRestService.class.getName());
+
+    private final String[] doNotCopyUser = {
```

Comment:
If we leave out the `user` field from the DTO, to only allow for it to be set internally within the backend and not from any external sources, then we don't need this.